db/commitlog/commitlog.cc

```diff
@@ -2253,6 +2265,9 @@ future<> db::commitlog::segment_manager::do_pending_deletes() {
         co_return;
     }
 
+    promise<> p;
+    auto f = std::exchange(_pending_deletes, p.get_future());
```

Comment:
```suggestion
    promise<> current_deleted;
    auto previous_deleted = std::exchange(_pending_deletes, current_deleted.get_future());
```

probably we could use better names than these single character names? as they have relatively large lexical scope than a 3-line for loop, better names would help with the readability, i think.

Reply:
Sure.